site/content/blog/2025-02-21-hipcheck-3.11.0-release.md

```diff
@@ -0,0 +1,198 @@
+a given analysis; we are in turn defining a plugin's "readinesss" as whether it
+successfully completes the configuration step. We encourage plugin developers to
+use this step to do all necessary file parsing and environment checking, and not
+to report configuration success unless your criteria are met.
+
+#### Better Configuration Error Reporting
+
+With the above changes, `hc ready` now informs us if the configuration of a
+plugin (based on the supplied policy file) failed. To help users better
+understand and address configuration errors that may arise, we made the
+following additional changes. Firstly, for transparency Hipcheck now prints the
+name and version of the plugins that failed configuration. We also expanded the
+range of configuration error types that plugins can report, so that plugin
+authors can be more clear to users about what caused the failure. This change is
+reflected in the `ConfigurationStatus` enum, part of our gRPC protocol
+definition at `hipcheck-common/proto/hipcheck/v1/hipcheck.proto`. The new errors
+are as follows:
+
+```
+// The plugin encountered an internal error, probably due to incorrect assumptions.
+CONFIGURATION_STATUS_INTERNAL_ERROR = 5;
+
+// A necessary plugin input file was not found.
+CONFIGURATION_STATUS_FILE_NOT_FOUND = 6;
+
+// The plugin's input data could not be parsed correctly.
+CONFIGURATION_STATUS_PARSE_ERROR = 7;
+
+// An environment variable needed by the plugin was not set.
+CONFIGURATION_STATUS_ENV_VAR_NOT_SET = 8;
+
+// The plugin could not find or run a needed program.
+CONFIGURATION_STATUS_MISSING_PROGRAM = 9;
+```
+
+The suite of plugins maintained by the Hipcheck team has been updated to make
+use of these new error types where appropriate. Finally, using the `miette`
+crate we have updated our plugins that do file parsing to return high-fidelity
+parsing error messages that show where in the file the error arose and suggested
+fixes.
+
+### 📦&nbsp;&nbsp;JSON-formatted Analysis Reporting
+
+Users can select between having Hipcheck emit its analysis report in the shell
+or as a JSON file using the `--format` flag. Previously, the JSON format was not
+a information-parity with the shell output. With Hipcheck 3.11.0 we have
+addressed this deficiency and the two formats are mostly equal in
+expressiveness. The default JSON output (`--format json`) reports only an
+abridged version of each plugin's raw analysis output to prioritize report
+readability. If users are interested in a JSON output file with each plugin's
+raw output (i.e.  before converted to a pass/fail determination with the policy
+expression), they can use the new `--format debug` option.
+
+Internally, the `ReportBuilder` structure that Hipcheck previously used to build
+the shell output has been eliminated; now, shell output is produced using a
+[mini-jinja][mini_jinja] template applied to the JSON output, which is a much
+more elegant design.
```

Comment:
"which is a much more elegant design" -> "which simplifies the internals for report generation while also making future report improvements easier."